src/lightning/app/core/plugin.py

```diff
@@ -0,0 +1,167 @@
+
+        return resp.json()
+
+    def _setup(self, app_id: str) -> None:
+        client = LightningClient()
+        project_id = _get_project(client).project_id
+        response = client.lightningapp_instance_service_list_lightningapp_instances(
+            project_id=project_id, app_id=app_id
+        )
+        if len(response.lightningapps) > 1:
+            raise RuntimeError(f"Found multiple apps with ID: {app_id}")
+        if len(response.lightningapps) == 0:
+            raise RuntimeError(f"Found no apps with ID: {app_id}")
+        self.app_url = response.lightningapps[0].status.url
+
+
+class _Run(BaseModel):
+    plugin_name: str
+    project_id: str
+    cloudspace_id: str
+    name: str
+    entrypoint: str
+    cluster_id: Optional[str] = None
+    app_id: Optional[str] = None
+
+
+def _run_plugin(run: _Run) -> None:
+    """Create a run with the given name and entrypoint under the cloudspace with the given ID."""
+    if run.app_id is None and run.plugin_name == "app":
+        from lightning.app.runners.cloud import CloudRuntime
+
+        # TODO: App dispatch should be a plugin
+        # Dispatch the run
+        _set_flow_context()
+
+        entrypoint_file = Path("/content") / run.entrypoint
+
+        app = CloudRuntime.load_app_from_file(str(entrypoint_file.resolve().absolute()))
+
+        app.stage = AppStage.BLOCKING
+
+        runtime = CloudRuntime(
+            app=app,
+            entrypoint=entrypoint_file,
+            start_server=True,
+            env_vars={},
+            secrets={},
+            run_app_comment_commands=True,
+        )
+        # Used to indicate Lightning has been dispatched
+        os.environ["LIGHTNING_DISPATCHED"] = "1"
+
+        try:
+            runtime.cloudspace_dispatch(
+                project_id=run.project_id,
+                cloudspace_id=run.cloudspace_id,
+                name=run.name,
+                cluster_id=run.cluster_id,
+            )
+        except Exception as e:
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
+    elif run.app_id is not None:
+        from lightning.app.utilities.cli_helpers import _LightningAppOpenAPIRetriever
+        from lightning.app.utilities.commands.base import _download_command
+
+        retriever = _LightningAppOpenAPIRetriever(run.app_id)
+
+        metadata = retriever.api_commands[run.plugin_name]  # type: ignore
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+
+            target_file = os.path.join(tmpdir, f"{run.plugin_name}.py")
+            plugin = _download_command(
+                run.plugin_name,
+                metadata["cls_path"],
+                metadata["cls_name"],
+                run.app_id,
+                target_file=target_file,
+            )
+
+            if isinstance(plugin, Plugin):
+                plugin._setup(app_id=run.app_id)
+                plugin.run(run.name, run.entrypoint)
```

Comment:
Would it make sense to run this in a separate env to isolate dependencies? We could use venv-spawn if needed (https://gist.github.com/lantiga/4273e893eb50232840fcf221aa4afcbb)

Reply:
That's a nice suggestion. Will look into it in a follow-up - the only possible issue is that the server will run from a sidecar with minimal resources so this may get too heavy for it